src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -114,26 +119,142 @@ No config file found. Your Okta session will not be cached.
 		[NotNullWhen( true )] out IOktaAuthenticatedClient? oktaAuthenticated
 	) {
-		string? sessionId = GetCachedOktaSessionId( user, org );
+		string? sessionId = GetCachedOktaSessionId( user, orgBaseAddress.Host );
 		if( string.IsNullOrEmpty( sessionId ) ) {
 			oktaAuthenticated = null;
 			return false;
 		}
 
-		oktaAuthenticated = oktaClientFactory.CreateAuthenticatedClient( org, sessionId );
+		oktaAuthenticated = oktaClientFactory.CreateAuthenticatedClient( orgBaseAddress, sessionId );
 		return true;
 	}
 
+	private async Task<IOktaAuthenticatedClient?> GetDssoAuthenticatedClientAsync(
+		Uri orgUrl,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool nonInteractive,
+		bool experimentalBypassBrowserSecurity
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimentalBypassBrowserSecurity );
+		if( browser is null ) {
+			return null;
+		}
+
+		if( !nonInteractive ) {
+			Console.Error.WriteLine( "Attempting to automatically login using Okta Desktop Single Sign-On." );
+		}
+		using var cancellationTokenSource = new CancellationTokenSource( TimeSpan.FromSeconds( 15 ) );
+		var sessionIdTcs = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		string? sessionId;
+
+		try {
+			using var page = await browser.NewPageAsync().WaitAsync( cancellationTokenSource.Token );
+			int attempt = 1;
+
+			page.Load += ( _, _ ) => _ = GetSessionCookieAsync();
+			await page.GoToAsync( orgUrl.AbsoluteUri ).WaitAsync( cancellationTokenSource.Token );
+			sessionId = await sessionIdTcs.Task;
+
+			async Task GetSessionCookieAsync() {
+				var url = new Uri( page.Url );
+				if( url.Host == orgUrl.Host ) {
+					string title = await page.GetTitleAsync().WaitAsync( cancellationTokenSource.Token );
+					// DSSO can sometimes takes more than one attempt.
+					// If the path is '/' with 'sign in' in the title, it means DSSO is not available and we should stop retrying.
+					if( title.Contains( "sign in", StringComparison.OrdinalIgnoreCase ) ) {
+						if( attempt < 3 && url.AbsolutePath != "/" ) {
+							attempt++;
+							await page.GoToAsync( orgUrl.AbsoluteUri ).WaitAsync( cancellationTokenSource.Token );
+						} else {
+							consoleWriter.WriteWarning(
+								"WARNING: Could not authenticate with Okta using Desktop Single Sign-On." );
+							sessionIdTcs.SetResult( null );
+						}
+						return;
+					}
+				}
+				var cookies = await page.GetCookiesAsync( orgUrl.AbsoluteUri ).WaitAsync( cancellationTokenSource.Token );
+				if( Array.Find( cookies, c => c.Name == "sid" )?.Value is string sid ) {
+					sessionIdTcs.SetResult( sid );
+				}
+			}
+		} catch( TaskCanceledException ) {
+			consoleWriter.WriteWarning( $"""
+				WARNING: Timed out when trying to create Okta session through Desktop Single Sign-On.
+				Check if the org '{orgUrl}' is correct. If running BMX with elevated privileges,
+				rerun the command with the '--experimental-bypass-browser-security' flag
```

Comment:
Please avoid mentioning DSSO.

I'd also rather not direct people to use `--experimental-bypass-browser-security` without more qualifications, e.g. 

> if you _have to_ run BMX with elevated privileges and you aren't concerned with the security of {orgUrl}, consider ...

Also, to confirm - admin mode without --no-sandbox triggers `TaskCanceledException`?
That seems surprising.

Reply:
You mentioned before that it's straightforward to detect elevated privileges in all OSes. Can we do that?
Can be in a separate PR though.

Reply:
I saw some stackoverflow posts about it but never tried implementing. I can try after yea

Reply:
> Also, to confirm - admin mode without --no-sandbox triggers TaskCanceledException?

Yea it can hang on `browser.NewPageAsync()` and sometimes would throw the other